### PR TITLE
Ensure the user exists before calling a method on it 

### LINKED ISSUE #24751: Public upload: Call to a member function getQuota() on null in lib/private/util.php on line 289
### Steps to reproduce
1. Allow public uploads in administrator options
2. Share a folder as link with "Allow editing"
3. Browse to shared link 
4. Try to upload a file

### Expected behaviour
Files successfully uploaded.

### Actual behaviour
Upload error message:
"Fehler beim Hochladen: "xyz.pdf": Internal Server Error

### Server configuration
**Operating system**: 
Debian Jessie
**Web server:**
nginx
**Database:**
mysql  Ver 14.14 Distrib 5.5.49
**PHP version:**
PHP 5.6.20-0+deb8u1 (cli)
**ownCloud version:** (see ownCloud admin page)
ownCloud 9.0.2 (stable)
**Updated from an older ownCloud or fresh install:**
Updated from 9.0
**Where did you install ownCloud from:**
owncloud.org
**Signing status (ownCloud 9.0 and above):**
```
No errors have been found.
```

**List of activated apps:**
```
Enabled:
  - activity: 2.2.1
  - comments: 0.2
  - dav: 0.1.6
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - external
  - files_external
  - user_external
  - user_ldap
```

**The content of config/config.php:**
```
{
    "system": {
        "updatechecker": false,
        "instanceid": "ocm7tfuygjh0",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.wdj.ac"
        ],
        "datadirectory": "\/media\/owncloud\/data",
        "overwrite.cli.url": "https:\/\/cloud.wdj.ac",
        "dbtype": "mysql",
        "version": "9.0.2.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "quota_include_external_storage": true,
        "asset-pipeline.enabled": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "mail_from_address": "hello",
        "mail_smtpmode": "php",
        "mail_domain": "wdj.ac",
        "loglevel": 1,
        "theme": "",
        "maintenance": false
    }
}
```

**Are you using external storage, if yes which one:** 
cifs
**Are you using encryption:** 
no
**Are you using an external user-backend, if yes which one:** no

#### Web server error log
```
2016/05/20 20:12:16 [error] 718#718: *20042 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Call to a member function getQuota() on null in /var/www/owncloud/lib/private/util.php on line 289" while reading response header from upstream, client: XXXXXX, server: cloud.wdj.ac, request: "POST /index.php/apps/files/ajax/upload.php HTTP/1.1", upstream: "fastcgi://unix:/var/run/cloud.wdj.ac.sock:", host: "cloud.wdj.ac"
```

#### ownCloud log (data/owncloud.log)
```
nothing abut this
```

#### Browser log
```
console log
POST https://cloud.wdj.ac/index.php/apps/files/ajax/upload.php 500 (Internal Server Error)
```

### PR DESCRIPTION
**Pull request by DeepDiver1975**
fixes #24751

stable9.1: https://github.com/owncloud/core/pull/25776
stable9: https://github.com/owncloud/core/pull/25777

### CODE REVIEW

File: lib/private/legacy/util.php [hunk 1]
```diff
@@ -293,12 +294,15 @@ public static function isDefaultExpireDateEnforced() {
 	 * @return int Quota bytes
 	 */
 	public static function getUserQuota($user) {
-		$userQuota = \OC::$server->getUserManager()->get($user)->getQuota();
+		$user = \OC::$server->getUserManager()->get($user);
```
Comment (VicDeo) [inline lib/private/legacy/util.php]: @DeepDiver1975 `$user` is reused. Rename argument to `$uid`?

Comment (DeepDiver1975) [inline lib/private/legacy/util.php]: I dont see an issue with this because the variable is not used further down

Comment (VicDeo) [inline lib/private/legacy/util.php]: @DeepDiver1975 not a real issue, I just used to read "$user" as "object" and "$uid" as "string" recently.

**Comment by dragotin** [comment]
looks :+1: apart from my question above.

**Comment by DeepDiver1975** [comment]
> If the user is not valid why return unlimited space? Wouldn't it be better to return zero?

unlimited was returned in this case from the very first day - see https://github.com/owncloud/core/commit/e2f04b3b429b42975f4457b77ca720396506e512#diff-9b514c1773091768aa8f61c5b6d74b9fR76

**Comment by PVince81** [comment]
Nice catch, I didn't know we still had other relics of this issue... :+1: 

Jenkins sleeping ?

File: lib/private/legacy/util.php (hunk 1 above)
Comment (butonic) [inline lib/private/legacy/util.php]: I made it `$userId` which is more readable to also trigger jenkins.

**Comment by butonic** [comment]
👍

**Comment by PVince81** [comment]
Test failure unrelated, merging

**Comment by PVince81** [comment]
@butonic please submit backports

**Comment by DeepDiver1975** [comment]
> @butonic please submit backports

I take care ....

**Pull request merged**